File: backend/db/models/project.js
"use strict";
const { Model } = require("sequelize");
module.exports = (sequelize, DataTypes) => {
  class Project extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      Project.hasMany(models.Step, { as: "steps", foreignKey: "projectId" });
    }
  }
  Project.init(
    {
      name: DataTypes.STRING(20),
      categoryId: {
        type: DataTypes.INTEGER,
        references: { model: "Categories" },
      },
      completed: {
        type: DataTypes.BOOLEAN,
        defaultValue: false,
      },
      blueprintId: {
        type: DataTypes.INTEGER,
        references: { model: "BluePrints" },
      },
    },
    {
      sequelize,
      modelName: "Project",
    }
  );
  return Project;
};

File: backend/app/models/project.py
from .db import db


class Project(db.Model):
    __tablename__ = 'projects'
    
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(50))
    completed = db.Column(db.Boolean, default=False)
    blueprint_id = db.Column(db.Integer, db.ForeignKey('blueprints.id'), nullable=False)
    category_id = db.Column(db.Integer, db.ForeignKey('categories.id'), nullable=False)
    created_at = db.Column(db.DateTime, default=db.func.now())
    updated_at = db.Column(db.DateTime, default=db.func.now(), onupdate=db.func.now())
    
    steps = db.relationship('Step', backref='projects', lazy=True)

    def to_dict(self):
        return {
            'id': self.id,
            'name': self.name,
            'completed': self.completed,
            'blueprint_id': self.blueprint_id,
            'category_id': self.category_id
        }
